worker/uniter/resolver/loop_test.go

```diff
@@ -176,7 +176,65 @@ func (s *LoopSuite) TestLoop(c *gc.C) {
 	c.Assert(err, gc.Equals, resolver.ErrLoopAborted)
 	c.Assert(resolverCalls, gc.Equals, 3)
 	s.executor.CheckCallNames(c, "State", "State", "Run", "State", "State")
-	c.Assert(s.executor.Calls()[2].Args, jc.SameContents, []interface{}{theOp})
+
+	runArgs := s.executor.Calls()[2].Args
+	c.Assert(runArgs, gc.HasLen, 2)
+	c.Assert(runArgs[0], gc.DeepEquals, theOp)
+	c.Assert(runArgs[1], gc.NotNil)
+}
+
+func (s *LoopSuite) TestLoopWithChange(c *gc.C) {
+	var resolverCalls int
+	theOp := &mockOp{}
+	s.resolver = resolver.ResolverFunc(func(
+		_ resolver.LocalState,
+		_ remotestate.Snapshot,
+		_ operation.Factory,
+	) (operation.Operation, error) {
+		resolverCalls++
+		switch resolverCalls {
+		// On the first call, return an operation.
+		case 1:
+			// queue up a change to trigger snapshot channel.
+			s.watcher.changes <- struct{}{}
+			return theOp, nil
+		// On the second call, simulate having
+		// no operations to perform, at which
+		// point we'll wait for a remote state
+		// change.
+		case 2:
+			s.watcher.changes <- struct{}{}
+			break
+		case 3:
+			break
+		// On the fourth call, kill the loop.
+		case 4:
+			close(s.abort)
+			break
+		}
+		return nil, resolver.ErrNoOperation
+	})
+
+	s.executor.run = func(op operation.Operation, rs <-chan remotestate.Snapshot) error {
+		select {
+		case _, ok := <-rs:
+			c.Assert(ok, jc.IsTrue)
```

Comment:
can we include a counter to check that we get the expected number of remote state watcher triggers?